src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/WaitRequestHandler.kt

```diff
@@ -187,37 +181,50 @@ class WaitRequestHandler(private val config: FrejaEidAuthenticatorPluginConfig)
                     _sessionManager.remove(SESSION_COUNTRY)
                 }
 
-                val customIdentifierAttribute = _sessionManager.get(SESSION_CUSTOM_IDENTIFIER)
-                if (customIdentifierAttribute != null)
-                {
-                    subjectAttributes.add(Attribute.of("custom-identifier", customIdentifierAttribute.value.toString()))
-                    _sessionManager.remove(SESSION_CUSTOM_IDENTIFIER)
-                }
-
                 val relyingPartyUserIdAttribute = _sessionManager.get(SESSION_RP_USER_ID)
                 if (relyingPartyUserIdAttribute != null)
                 {
                     subjectAttributes.add(Attribute.of("relyingPartyUserId", relyingPartyUserIdAttribute.value.toString()))
+                    subject = relyingPartyUserIdAttribute.value.toString()
```

Comment:
Not of a fan of all these `.value.toString()`. Since we controll both sides it's probably fine, but the value could be something else than a string.

Reply:
The value is a non-nullable `Comparable`, so `toString` will never fail at least.

Reply:
Right, never null, but could be another type than string, giving a weird result.